plugin-server/src/worker/ingestion/properties-updater.ts

```diff
@@ -152,45 +186,100 @@ export async function upsertGroup(
     }
 }
 
-export function calculateUpdate(
+export function shouldUpdateProperty(
+    operation: PropertyUpdateOperation,
+    timestamp: DateTime,
+    lastOperation: PropertyUpdateOperation,
+    lastTimestamp: DateTime
+): boolean {
+    if (
+        operation == PropertyUpdateOperation.SetOnce &&
+        lastOperation === PropertyUpdateOperation.SetOnce &&
+        lastTimestamp > timestamp
+    ) {
+        return true
+    }
+    if (
+        operation == PropertyUpdateOperation.Set &&
+        (lastOperation === PropertyUpdateOperation.SetOnce || lastTimestamp < timestamp)
+    ) {
+        return true
+    }
+    return false
+}
+
+export function calculateUpdateForMerge(
```

Comment:
there's a ton of overlap between this and `calculateUpdate`. Can we not have this do a bit of handling then pass things over to `calculateUpdate`?

Reply:
We can't do that, because `calculateUpdate` takes one timestamp only for the new properties, we could convert that input to this, but it's quite inefficient to do the conversion from `{Properties (set), Properties (setOnce), timestamp}` to `{Properties, PropertiesLastUpdatedAt, PropertiesLastOperation}` rather than what we have now. I can pull the `if` clause into a function though if that would help